src/Microsoft.PowerShell.Commands.Utility/commands/utility/Csv.cs

```diff
@@ -27,85 +29,59 @@ internal CSVHelper(char delimiter)
         /// String to be parsed.
         /// </param>
         internal Collection<string> ParseCsv(string csv)
-        {
+        {        
+            csv = csv.Trim(); 
             Collection<string> result = new Collection<string>();
-            string tempString = string.Empty;
-            csv = csv.Trim();
             if (csv.Length == 0 || csv[0] == '#')
             {
                 return result;
             }
 
-            bool inQuote = false;
-            for (int i = 0; i < csv.Length; i++)
+            var reader = new StringReader(csv);
+            StringBuilder wordBuffer = new StringBuilder();
+
+            while (reader.Peek() != -1) 
             {
-                char c = csv[i];
-                if (c == Delimiter)
+                char nextChar = (char)reader.Read();
+
+                // if next character was delimiter or we are at the end, add string to result and clear builder
+                // else if next character was quote, perform reading untill next quote and add it to builder
+                // else read and add it to builder
+                if (nextChar == Delimiter) 
                 {
-                    if (!inQuote)
-                    {
-                        result.Add(tempString);
-                        tempString = string.Empty;
-                    }
-                    else
-                    {
-                        tempString += c;
-                    }
-                }
-                else
+                    result.Add(wordBuffer.ToString());
+                    wordBuffer.Clear();
+                } 
+                else if (nextChar == '"') 
                 {
-                    switch (c)
+                    bool isinQuotes = true;
```

Comment:
Please replace isinQuotes with isInQuotes

Reply:
Unfortunately this raises a codeFactor error regarding Hungarian notation. I changed it to `betweenQuotes`

![image](https://user-images.githubusercontent.com/9801575/54298615-60fd7380-45b9-11e9-9e6b-a9cf76a7a8ba.png)